device_object.cpp

```diff
@@ -181,14 +181,7 @@ static const char *device_manufacturer(void) {
 }
 
 static const char *model_number(void) {
-#ifdef TARGET_DISCO_L496AG
-    return "STM32L496G-DISCO";
-#elif TARGET_NRF52840_DK
-    return "nRF52840 DK";
-#else
-#    warning "Add model number"
-    return "Unknown";
-#endif
+    return MBED_STRINGIFY(TARGET_NAME);
```

Comment:
This is a nice idea, but what do you think about retaining the names that are already there?

Something like:
```suggestion
#ifdef TARGET_DISCO_L496AG
    return "STM32L496G-DISCO";
#elif TARGET_NRF52840_DK
    return "nRF52840 DK";
#else
#    warning "Unsupported board, using TARGET_NAME as model number"
    return MBED_STRINGIFY(TARGET_NAME);
#endif
```
The default target names look more like identifiers than like actual board names, so I think it would make sense to use your suggestion more as a fallback rather than as the only option.

Reply:
The name difference is really small for those vs. what you have there as string. Literally its DISCO_L496AG vs. STM32L496G-DISCO. This one supports the whole range of Mbed OS supported boards in one go with no one maintaining anything in a list. 

In real life if anyone would go to production they would pick that up from a .h/.c file OR read it from KVStore.

This also works for custom targets.

But, your app - you do as you please.